src/environments/array.js

```diff
@@ -4,10 +4,12 @@ import Style from "../Style";
 import defineEnvironment from "../defineEnvironment";
 import {parseCD} from "./cd";
 import defineFunction from "../defineFunction";
+import defineMacro from "../defineMacro";
 import mathMLTree from "../mathMLTree";
 import ParseError from "../ParseError";
 import {assertNodeType, assertSymbolNodeType} from "../parseNode";
 import {checkSymbolNodeType} from "../parseNode";
+import {Token} from "../Token";
 import {calculateSize} from "../units";
 import utils from "../utils";
 
```

Comment:
```suggestion

const getAutoTag = name => name.indexOf("ed") === -1 ? name.indexOf("*") === -1
    : undefined;
```
(below `validateAmsEnvironmentContext`)